src/runtime/pkg/utils/utils.go

```diff
@@ -80,3 +80,16 @@ func EnsureDir(path string, mode os.FileMode) error {
 
 	return nil
 }
+
+func FirstValidPath(paths []string) (string, error) {
+	for _, p := range paths {
+		if _, err := os.Stat(p); err != nil {
+			if os.IsNotExist(err) {
```

Comment:
Nit: There is a small chance that an admin may have disabled execute permission on one of the (valid) binaries so although it exists, it won't work. That would defeat this function so can you check that the file is also executable?